components/data-table/__tests__/data-table.browser-test.jsx

```diff
@@ -248,6 +248,26 @@ describe('DataTable: ', function describeFunction() {
 			checkedBoxes.should.have.length(1);
 		});
 
+		it('can start with a row disabled', function () {
+			renderTable(
+				<DataTable
+					{...defaultProps}
+					selection={[]}
+					disabledSelection={defaultSelection}
+				>
+					{columns.map((columnProps) => (
+						<DataTableColumn {...columnProps} key={columnProps.property} />
+					))}
+				</DataTable>
+			).call(this);
+
+			const tbody = getTable(this.dom).querySelectorAll('tbody')[0];
+			const disabledRows = tbody.querySelectorAll(
+				'.slds-checkbox input:disabled'
+			);
+			disabledRows.should.have.length(1);
```

Comment:
We should spy the `onRowChange` and make sure it doesn't fire OR make sure the data.selection parameter is still `0` in length. See `can select a row` below and negate it.

Reply:
updated test